lib/net.js

```diff
@@ -631,10 +631,20 @@ Socket.prototype.destroySoon = function() {
   if (this.writable)
     this.end();
 
-  if (this.writableFinished)
+  if (this.writableFinished && this.readableEnded)
     this.destroy();
   else
-    this.once('finish', this.destroy);
+    this.once('finish', () => {
+      // If we destroy the socket before we have received an incoming EOF
+      // the remote might still be sending data that will trigger a RST
+      // from this host once it is received
+      // Closing a TCP socket requires two FIN/ACK exchanges, one in every
+      // direction - this means two events
+      if (this.readableEnded)
+        this.destroy();
+      else
+        this.once('end', this.destroy);
+    });
```

Comment:
Is this change to `net` still needed?

Reply:
IMO these changes should be removed from the PR.

Reply:
I don't know. I wonder if this function should be modified, should be deleted or should be left in its current state.
It is not used anywhere else, but it is callable from user code even it is not supposed to be public.

Reply:
Please don’t change it. You can open a separate PR for runtime deprecation.